src/db/entity/ExecutionEntity.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  ManyToMany,
+  OneToMany,
+} from 'typeorm';
+import ConnectionEntity from './ConnectionEntity';
+import RuleEntity from './RuleEntity';
+
+/**
+ * A list of previous executions. Contains the test data, group, and test conditions.
+ * Currently supports only one database // TODO add support
+ */
+@Entity({ name: 'Execution' })
+class ExecutionEntity {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  timestamp: Date;
+
+  /**
+   * A list of the rules being tested
+   */
+  @OneToMany((_type) => RuleEntity, (rule) => rule)
+  rules: RuleEntity[];
+
+  /**
+   * The server connections to execute on
+   */
+  @ManyToMany(
+    (_type) => ConnectionEntity,
+    (connection) => connection.executions,
+    {
+      onDelete: 'CASCADE',
```

Comment:
You also might want to add on update cascades everywhere: https://www.sqlite.org/foreignkeys.html#:~:text=The%20ON%20DELETE%20and%20ON,defaults%20to%20%22NO%20ACTION%22.
There should be some way to just easily configure this globally

Reply:
Never mind there isn't, might just have to on update cascades everywhere